### PR TITLE
Update BacktrackingTM API docs

### PR DESCRIPTION
**Pull request by rhyolight**
Contributes to https://github.com/numenta/nupic/issues/3642.

### CODE REVIEW

File: src/nupic/algorithms/backtracking_tm_cpp.py [hunk 1]
```diff
@@ -713,77 +649,43 @@ def getSegmentInfo(self, collectActiveData = False):
...
-    assert False
-
-
-  def getLeastAllocatedCell(self, c):
-    """For the given column, return the cell with the fewest number of
-    segments."""
-
-    # TODO: add C++ accessor to implement this or implement our own variation
-    assert False
-
```
Comment (rhyolight) [inline src/nupic/algorithms/backtracking_tm_cpp.py]: I removed these three functions. They were never used, contained no logic, and did not seem to override anything.

File: src/nupic/algorithms/backtracking_tm_cpp.py [hunk 2]
```diff
@@ -517,13 +484,13 @@ def printSegment(self, s):
...
     assert False
-    print "=== SEGMENT UPDATES ===, Num = ",len(self.segmentUpdates)
+    print "=== SEGMENT UPDATES ===, Num = ", len(self.segmentUpdates)
     for key, updateList in self.segmentUpdates.iteritems():
       c,i = key[0],key[1]
       print c,i,updateList
 
 
-  def slowIsSegmentActive(self, seg, timeStep):
+  def _slowIsSegmentActive(self, seg, timeStep):
```
Comment (rhyolight) [inline src/nupic/algorithms/backtracking_tm_cpp.py]: I made this function private, because it is only called from `printCell()`.

**Comment by rhyolight** [comment]
@scottpurdy or @oxtopus Mostly doc changes, but I cleaned up some code (see my comments).

File: src/nupic/algorithms/backtracking_tm_cpp.py [hunk 3]
```diff
@@ -406,34 +394,13 @@ def reset(self):
...
-    trimSegments. (finishLearning is here for backward compatibility)
-    """
     # Keep weakly formed synapses around because they contain confidence scores
     #  for paths out of learned sequenced and produce a better prediction than
     #  chance.
     self.trimSegments(minPermanence=0.0001)
 
 
   def trimSegments(self, minPermanence=None, minNumSyns=None):
-    """This method deletes all synapses where permanence value is strictly
```
Comment (scottpurdy) [inline src/nupic/algorithms/backtracking_tm_cpp.py]: why delete this doc string?

File: src/nupic/algorithms/backtracking_tm_cpp.py [hunk 4]
```diff
@@ -577,31 +544,23 @@ def printCell(self, c, i, onlyActiveSegments=False):
...
+    :param idx: (int) global cell index
+    :returns: (tuple) (colIdx, cellIdxInCol)
     """
     c = idx//self.cellsPerColumn
     i = idx - c*self.cellsPerColumn
     return c,i
 
 
   def getSegmentOnCell(self, c, i, segIdx):
-    """Return segment number segIdx on cell (c,i).
```
Comment (scottpurdy) [inline src/nupic/algorithms/backtracking_tm_cpp.py]: why delete this doc string?

File: src/nupic/algorithms/backtracking_tm_cpp.py [hunk 5]
```diff
@@ -406,34 +394,13 @@ def reset(self):
 
 
   def finishLearning(self):
-    """Called when learning has been completed. This method just calls
```
Comment (scottpurdy) [inline src/nupic/algorithms/backtracking_tm_cpp.py]: why delete this doc string?

File: src/nupic/algorithms/backtracking_tm_cpp.py [hunk 6]
```diff
@@ -394,10 +386,6 @@ def _setStatePointers(self):
 
 
   def reset(self):
-    """ Reset the state of all cells.
```
Comment (scottpurdy) [inline src/nupic/algorithms/backtracking_tm_cpp.py]: why delete this doc string?

File: src/nupic/algorithms/backtracking_tm_cpp.py [hunk 7]
```diff
@@ -278,12 +280,6 @@ def __getattr__(self, name):
 
 
   def compute(self, bottomUpInput, enableLearn, enableInference=None):
-    """ Handle one compute, possibly learning.
```
Comment (scottpurdy) [inline src/nupic/algorithms/backtracking_tm_cpp.py]: why delete this doc string?

**Comment by rhyolight** [comment]
@scottpurdy All the docstring deletions you mentioned are actually duplicate docstrings. All these functions are overridden, and simply copy/pasted the docstrings from the super function.

I find docs much cleaner without the duplication, and the docs are structured in a way that you can navigate from the extension class to the base class easily.

**Comment by rhyolight** [comment]
I'm going to put in links from the overridden function docstrings to the super function docs just to make it crystal clear.

**Pull request merged**